flutter/lib/src/span_frame_metrics_collector.dart

```diff
@@ -0,0 +1,226 @@
+import 'dart:collection';
+
+import 'package:flutter/cupertino.dart';
+import 'package:meta/meta.dart';
+
+import '../sentry_flutter.dart';
+
+import 'frame_callback_handler.dart';
+import 'native/sentry_native_binding.dart';
+
+@internal
+class SpanFrameMetricsCollector implements PerformanceContinuousCollector {
+  static const _frozenFrameThresholdMs = 700;
+  static const totalFramesKey = 'frames.total';
+  static const framesDelayKey = 'frames.delay';
+  static const slowFramesKey = 'frames.slow';
+  static const frozenFramesKey = 'frames.frozen';
+
+  final SentryFlutterOptions options;
+  final FrameCallbackHandler _frameCallbackHandler;
+  final SentryNativeBinding? _native;
+
+  final bool _isTestMode;
+
+  /// Stores frame timestamps and their durations in milliseconds.
+  /// Keys are frame timestamps, values are frame durations.
+  final frameDurations = SplayTreeMap<DateTime, int>();
+
+  /// Stores the spans that are actively being tracked.
+  /// After the frames are calculated and stored in the span the span is removed from this list.
+  final activeSpans = SplayTreeSet<ISentrySpan>(
+      (a, b) => a.startTimestamp.compareTo(b.startTimestamp));
+
+  bool get isTrackingPaused => _isTrackingPaused;
+  bool _isTrackingPaused = true;
+
+  bool get isTrackingRegistered => _isTrackingRegistered;
+  bool _isTrackingRegistered = false;
+
+  int displayRefreshRate = 60;
+
+  final _stopwatch = Stopwatch();
+
+  SpanFrameMetricsCollector(this.options,
+      {FrameCallbackHandler? frameCallbackHandler,
+      SentryNativeBinding? native,
+      bool isTestMode = false})
+      : _frameCallbackHandler =
+            frameCallbackHandler ?? DefaultFrameCallbackHandler(),
+        _native = native ?? SentryFlutter.native,
+        _isTestMode = isTestMode;
+
+  @override
+  Future<void> onSpanStarted(ISentrySpan span) async {
+    if (span is NoOpSentrySpan || !options.enableFramesTracking) {
+      return;
+    }
+
+    final fetchedDisplayRefreshRate = await _native?.displayRefreshRate();
+    if (fetchedDisplayRefreshRate != null) {
+      options.logger(SentryLevel.info,
```

Comment:
I think info would be too spammy for the users
```suggestion
      options.logger(SentryLevel.debug,
```